misc/rapiddisk-rootdev/ubuntu/rapiddisk_sub.orig

```diff
@@ -1,9 +1,13 @@
 #!/bin/sh
 
+. /scripts/functions
+
+modprobe -v dm-writecache
```

Comment:
I wonder if we should check in /sys/module to see if they are loaded afterward and if not, abort.
Maybe do not abort if dm-writecache is not loaded and the user did not specify a wb policy.

Reply:
I would love to push these updates into master. Any thoughts on my comment here and the one below?

Reply:
> I wonder if we should check in /sys/module to see if they are loaded afterward and if not, abort.
> Maybe do not abort if dm-writecache is not loaded and the user did not specify a wb policy.

> I would love to push these updates into master. Any thoughts on my comment here and the one below?

I have a new version ready, which implements your suggestions. My laptop exploded and I have to look for it in backups. If you can wait some more time I'll commit it.

Reply:
Awesome. Not the laptop exploding part. That part sucks and I am sorry about that. Whenever you can commit is ok with me. No rush.

Reply:
> @matteotenca
"But if I create a first ramdisk (rd0), an then a second one (rd1), the latter attaches without problem to the boot device. I know this is just a workaround: do you have any clue?"

The same had happened here ... the solution was when running the initramfs script configure the default cache to another partition not being root.
Thus I had created a partition 1 MB empty only to point the configuration.
The first cache being wa to a partition empty.
I not have information to explain because the issue was fixed doing it.
Need avoid create both wa and wb caches in same partition root ?

Reply:
> 

Any chance in adding a scipt to flush all writeback cache in OS restart and shutdown in rapiddisk ?

Reply:
@Augusto7743 This could be another pull request, maybe tied to one of your tickets. Remember, RapidDisk does not officially support dm-writecache. The option is there but a "use at your own risk" option. The extent of what I am willing to support is likely an init/systemd script to scan for all dm-writecache enabled devices and flush them on shutdown. The user would have to follow instructions to manually enable it.